src/Silverback.Integration/Messaging/Outbound/TransactionalOutbox/OutboxProduceStrategy.cs

```diff
@@ -46,7 +45,10 @@ public Task ProduceAsync(IOutboundEnvelope envelope)
 
                 _logger.LogWrittenToOutbox(envelope);
 
-                _producer ??= _outboundQueueBroker.GetProducer(envelope.Endpoint);
+                if (_producer == null || _producer.Endpoint.Name != envelope.Endpoint.Name)
+                {
+                    _producer = _outboundQueueBroker.GetProducer(envelope.Endpoint);
```

Comment:
Just to be sure that this is thread-safe, the created producer should be set to a local variable as well. The local variable should be used on line 53 instead of the field.

Reply:
Updated